src/coreclr/tools/Common/Compiler/TypeParser.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Internal.TypeSystem;
+using System.Diagnostics;
+
+namespace ILCompiler
+{
+    public class TypeParser
```

Comment:
Could you name it in a way that puts `Cecil` in the name? `CecilCompatibleTypeParser` or something like that. It should be clear that we don't define the parsing or escaping rules. Cecil does.

Reply:
We could also just inline it as a nested class into ProcessLinkerXmlBase. I'm not aware of a different spot where we would need it.